store/CSDK/test/main.cpp

```diff
@@ -210,6 +212,288 @@ bool tryCatchException(JNIEnv *env) {
+
+            jobject boolField = env->NewStringUTF("true");
+            env->SetObjectArrayElement(arr, 5, boolField);
+
+            jobject dateField = env->NewStringUTF(" 2019-03-02");
+            env->SetObjectArrayElement(arr, 6, dateField);
+
+            jobject timeField = env->NewStringUTF("2019-02-12 03:03:34");
+            env->SetObjectArrayElement(arr, 7, timeField);
+
+            char ctrFloat[10];
+            gcvt(floatValue, 10, ctrFloat);
+            floatValue = floatValue + 2;
+            jobject floatField = env->NewStringUTF(ctrFloat);
+            env->SetObjectArrayElement(arr, 8, floatField);
+
+            jobject arrayField = env->NewStringUTF("Hello#World#From#Carbon");
+            env->SetObjectArrayElement(arr, 9, arrayField);
+
+
+            writer.write(arr);
+
+            env->DeleteLocalRef(stringField);
+            env->DeleteLocalRef(shortField);
+            env->DeleteLocalRef(intField);
+            env->DeleteLocalRef(longField);
+            env->DeleteLocalRef(doubleField);
+            env->DeleteLocalRef(floatField);
+            env->DeleteLocalRef(dateField);
+            env->DeleteLocalRef(timeField);
+            env->DeleteLocalRef(boolField);
+            env->DeleteLocalRef(arrayField);
+            env->DeleteLocalRef(arr);
+        }
+        writer.close();
+
+        CarbonReader carbonReader;
+        carbonReader.builder(env, path);
+        carbonReader.build();
+        int i = 0;
+        CarbonRow carbonRow(env);
+        while (carbonReader.hasNext()) {
+            jobject row = carbonReader.readNextRow();
+            i++;
+            carbonRow.setCarbonRow(row);
+            printf("%s\t%d\t%ld\t", carbonRow.getString(0), carbonRow.getInt(1), carbonRow.getLong(2));
+            jobjectArray array1 = carbonRow.getArray(3);
+            jsize length = env->GetArrayLength(array1);
+            int j = 0;
+            for (j = 0; j < length; j++) {
+                jobject element = env->GetObjectArrayElement(array1, j);
+                char *str = (char *) env->GetStringUTFChars((jstring) element, JNI_FALSE);
+                printf("%s\t", str);
+            }
+            printf("%d\t", carbonRow.getShort(4));
+            printf("%d\t", carbonRow.getInt(5));
+            printf("%ld\t", carbonRow.getLong(6));
+            printf("%lf\t", carbonRow.getDouble(7));
+            bool bool1 = carbonRow.getBoolean(8);
+            if (bool1) {
+                printf("true\t");
+            } else {
+                printf("false\t");
+            }
+            printf("%f\t\n", carbonRow.getFloat(9));
+            env->DeleteLocalRef(row);
+        }
+        carbonReader.close();
+    } catch (jthrowable ex) {
+        env->ExceptionDescribe();
+        env->ExceptionClear();
+    }
+}
+
+/**
+ * test write data to S3
+ *
+ * @param env  jni env
+ * @return
+ */
+bool writeToS3(JNIEnv *env, char *path, char *argv[]) {
```

Comment:
Just in FAQ document, add how to use writer and reader with S3a environment. In that just Mention about withHadoopConf() API .
No need to add each time test case for s3 and normal. Only difference between this s3 test case and above testcase is calling withHadoopConf(). 

@KanakaKumar @jackylk  check this.

I see lot of duplicate test code for reader and writer in the name of s3 example. This is Redundant and to maintain and refactor in future.

Reply:
removed writeToS3.